assets/js/vue-apps/form-components/budget-input.vue

```diff
@@ -41,27 +53,36 @@ export default {
                 if (this.shouldAddNewRow()) {
                     this.addRow();
                 }
-                this.error.TOO_MANY_ITEMS =
-                    this.budgetRows.length === this.maxItems;
-                this.error.OVER_BUDGET =
-                    this.maxBudget && this.total > this.maxBudget;
-
-                if (this.error.OVER_BUDGET) {
-                    trackEvent('Budget Component', 'Error', 'Over budget');
-                }
-
-                if (this.error.TOO_MANY_ITEMS) {
-                    trackEvent(
-                        'Budget Component',
-                        'Error',
-                        'Maximum number of items reached'
-                    );
-                }
+                this.checkErrors();
             },
             deep: true
         }
     },
     methods: {
+        checkErrors() {
+            Vue.set(
+                this.error,
+                'TOO_MANY_ITEMS',
+                this.budgetRows.length === this.maxItems
+            );
+            Vue.set(
```

Comment:
Had to use this rather than the old `this.error.TOO_MANY_ITEMS = true` syntax as this wasn't properly reactive when called in this way (eg. at pageload)

Reply:
Hmm, this sounds like we should be setting an initial value for this in the `data()` set up step, then this could use the regular `this.` syntax.

Something like

```js
data() {
        // Add a ready-to-use new row if the budget isn't over the limit already
        const initialBudgetRows =
            this.budgetData.length < this.maxItems
                ? concat(this.budgetData, [{ item: '', cost: '' }])
                : this.budgetData;

        return {
            budgetRows: initialBudgetRows,
            error: {
                TOO_MANY_ITEMS: initialBudgetRows.length === this.maxItems,
                OVER_BUDGET: this.maxBudget && this.total > this.maxBudget
            }
        };
    },
```